### ISSUE TITLE
Color doesn't changed after installed the plugin

### ISSUE THREAD

**Issue by Avimitin** [Iron-E/nvim-highlite#11]
I have used the `nvim-highlite` template to build a color scheme `deus`. And after follow all the steps, and import the plugin by vim-plug. The color schemes look like no effect. And I try other themes like the `vim-one`, they are still working.

Could you please take a look at my configuration to check if there were any errors?

Here is my [theme](https://github.com/Avimitin/neovim-deus)
And here is my [neovim configuration](https://github.com/Avimitin/nvim).

**Comment by Avimitin** [comment]
My neovim version:

```text
NVIM v0.5.0-dev+1390-g988f88c24
```

By the way, the `nvim-highlite` plugin takes no effect with the below setting too.

```vimscript
Plug 'Iron-E/nvim-highlite'

color highlite
```

**Comment by Iron-E** [comment]
Hey, thanks for trying out this repository!

A cursory look at what you have indicates that your theme is named "deus". Because of that, running `colorscheme highlite` won't have any effect. Use `colorscheme deus` instead

I'll clone these repos and try to confirm later 👍

**Comment by Avimitin** [comment]
> Hey, thanks for trying out this repository!
> 
> A cursory look at what you have indicates that your theme is named "deus". Because of that, running `colorscheme highlite` won't have any effect. Use `colorscheme deus` instead
> 
> I'll clone these repos and try to confirm later 👍
> 
> But you can't get it to show up with the base repo either? That's interesting… I'll investigate that too. Normally you have to `Plug` your local repository while working on it but if you're trying with this upstream repo that shouldn't be an issue

Ah, I try the `colorscheme deus` setting, it doesn't take effect. That's why I try the base repository. But both of them don't work.

 It's weird that other themes like vim-one and gruvbox (which I temporarily use it as a replacement of the broken theme) are work. But I lack of programming experience of Lua, the plugin have no error messages too. I can't figure out what causes this problem.

You can try the latest build of the neovim if you have time.  The theme broke after I built the newer neovim. So i guess if the neovim have some new change that can't activate the theme.

**Comment by Iron-E** [comment]
I just updated Neovim to HEAD (`NVIM v0.5.0-dev+1385-g93f15db5d`) and downloaded your config repository to try out. Changing `color gruvbox` to `color deus` successfully shows a different colorscheme for me

I don't see that Neovim version (1390) on HEAD. The latest prerelease seems to be [1386](https://github.com/neovim/neovim/releases/tag/nightly)— can you try rebuilding to 1385 and seeing what happens?

Also, what exactly happens when you run `color deus`? Nothing or does an error popup?

**Comment by Avimitin** [comment]
# About the version

I built the latest neovim from [neovim-git](https://aur.archlinux.org/packages/neovim-git/), which checkout the [988f88c24](https://github.com/neovim/neovim/tree/988f88c24e83c37e1dc011e3561fb505ca7b8706) yesterday. 

---

I've tried the release version:

```console
$ ./nvim --version
NVIM v0.5.0-dev+1386-gfd7e77b22
```
which the theme is not working too.

---

Also, I tried checkout the `93f15db5d` commit and built the neovim:

```console
$ nvim --version
NVIM v0.5.0-dev+1385-g93f15db5d
```

It still can't work.

# About what happen

The theme will not report any errors. I've tried `:message` command but nothing reported. When I put the `color deus` in init script the Neovim just looks like default themes.

![图片](https://user-images.githubusercontent.com/30021675/121616056-bffe0b80-ca94-11eb-8e38-24217cd38862.png)

And I found another weird thing. If I put `color gruvbox` in the init script and use the command `:color deus`. The themes will work well. But if I just put it into the init script, the theme is still not working. I record a video at below.


https://user-images.githubusercontent.com/30021675/121619727-9e545280-ca9b-11eb-94c5-019b9a57bda3.mp4

**Comment by Avimitin** [comment]
Seems like the video is broken, I will describe the behavior of the neovim.

The environment I used.

```console
$ nvim --version
NVIM v0.5.0-dev+1385-g93f15db5d
```
![图片](https://user-images.githubusercontent.com/30021675/121625345-e0829180-caa5-11eb-95a9-1c6a443d5cc0.png)

I assign the `color gruvbox` in the init script first. Everything works fine. And in that neovim instance, I called the command: `:color deus`. The color scheme works as expect. 

![图片](https://user-images.githubusercontent.com/30021675/121625401-0019ba00-caa6-11eb-85ca-8863831257ba.png)

And I modified the init script, assigned `color deus` into it, exit the current neovim instance.

![图片](https://user-images.githubusercontent.com/30021675/121625473-1aec2e80-caa6-11eb-8270-e279d26d4562.png)

After reopening the neovim, the neovim instance now just used the default theme as the screenshot I captured below. 

![图片](https://user-images.githubusercontent.com/30021675/121625529-36efd000-caa6-11eb-8b6b-369db4f96e01.png)

And I called `:color gruvbox` command in this instance, the theme changed immediately as expected. 

![图片](https://user-images.githubusercontent.com/30021675/121625588-52f37180-caa6-11eb-97b7-fb54dffce21f.png)

But nothing changed except font color after I called the `:color deus` in this instance too.

![图片](https://user-images.githubusercontent.com/30021675/121625607-5e469d00-caa6-11eb-9521-aec33032476e.png)

**Comment by Iron-E** [comment]
So if I understand correctly, calling `:color deus` _does_ work but only _after_ Neovim has already initialized. Further, if Neovim attempts to load deus during initialization, calling `:color deus` doesn't work as expected.

Do I understand correctly?

> [Default](https://user-images.githubusercontent.com/30021675/121616056-bffe0b80-ca94-11eb-8e38-24217cd38862.png)

This is actually what I saw when attempting to use `:color deus` using your config. I'll do some more testing, I suspect it might be something in the configuration but I will try to use deus in my config as well to make sure.

---

Thanks again for your patience while I investigate this!

**Comment by Avimitin** [comment]
First of all, thanks for your reply!

The deus theme does work but only works after the neovim initialized with other themes. And if all things run as expect, the neovim should look like below.

![That's how deus looks like](https://raw.githubusercontent.com/ajmwagar/vim-deus/master/screencaps/vim-deus.jpg)

And yes, move the setting `set termguicolors` forward does take effect. Thanks for your helps!
If you need any other environment or setting that I can provide I will do my best to help you.

**Comment by Iron-E** [comment]
Can you try this branch and tell me whether or not it has been fixed?

```vim
Plug 'Iron-E/nvim-highlite', { 'branch': ' bugfix/001--change-tgc-during-runtime' }
```

**Comment by Avimitin** [comment]
I am sorry that this branch is still not working. The theme is still working only after setting the `termguicolors` option after the color set.

**Comment by Iron-E** [comment]
I accidentally had a typo in the instructions. Can you remove the old one from your config, then `:PlugClean`. Add the above repo + branch back, then:

```vim
:PlugInstall
:PlugUpdate
```

When I downloaded your config again just now to test it, I had to `:PlugUpdate` again before it worked. After that I was able to `:set termguicolors` after `:colorscheme highlite` and it worked.

**Comment by Avimitin** [comment]
Yes, it does works. Thanks for your help. Should I just use this branch to build a new theme or wait for you to merge it into master?

**Comment by Iron-E** [comment]
I will merge into master tomorrow. Use the workaround for now!

**Comment by Iron-E** [comment]
The update has been merged into master. Feel free to pull in the changes to your theme!
